src/rules/prettier/utils/parse-vue.ts

```diff
@@ -48,12 +48,13 @@ const processSFCBlock = ({
   loc,
   type,
 }: SFCBlock): PrettierVueSFCBlock => {
-  if (['template', 'script', 'style'].includes(type)) {
+  if (['template', 'script', 'scriptSetup', 'style'].includes(type)) {
     // `<template>`, `<script>`, `<style>` blocks should be treated as `.vue` file
     // to make the `vueIndentScriptAndStyle` option of prettier works, so we need
     // to wrap them with their original tags
-    const startingTag = `<${type}${lang ? ` lang="${lang}"` : ''}>`;
-    const endingTag = `</${type}>\n`;
+    const tagName = type === 'scriptSetup' ? 'script setup' : type;
+    const startingTag = `<${tagName}${lang ? ` lang="${lang}"` : ''}>`;
+    const endingTag = `</${tagName}>\n`;
```

Comment:
Then the ending tag would be `</script setup>`

Reply:
Just add an extra condition like this:

```
const startingTag = `<${type}${type === 'scriptSetup' ? 'setup' : ''}${lang ? ` lang="${lang}"` : ''}>`;
```

Reply:
Fixed.